### PR TITLE
Use rustfmt

### PR DESCRIPTION
**Pull request by dtolnay**
Just to see what this would look like. Let's wait for #437 to merge before moving forward with this so that we avoid merge conflicts. This PR is much easier to rebase.

I have not read through this carefully but I am making a PR so we can discuss the formatting and file rustfmt issues as necessary. I already filed https://github.com/rust-lang-nursery/rustfmt/issues/1094.

One thing that got much worse is nested aster builders. Before:

```rust
    let wrapper_ty = builder.path()
        .segment("__SerializeWith")
            .with_generics(wrapper_generics.clone())
            .build()
        .build();
```

After:

```rust
    let wrapper_ty = builder.path()
        .segment("__SerializeWith")
        .with_generics(wrapper_generics.clone())
        .build()
        .build();
```

Ref #438.

### CODE REVIEW

File: testing/tests/test_macros.rs [hunk 1]
```diff
@@ -39,91 +34,81 @@ struct DeNamedMap<A, B, C> {
...
-        e: &'a mut D,
-        //f: <D as Trait>::Type,
+        // d: C::Type,
+        e: &'a mut D, // f: <D as Trait>::Type,
     },
 }
 
 #[derive(Debug, PartialEq, Serialize, Deserialize)]
-enum DeEnum<B, C: /* Trait */, D> /* where D: Trait */ {
+enum DeEnum<B, C /* : /* Trait */, D> {
```
Comment (dtolnay) [inline testing/tests/test_macros.rs]: Well this is a bug.

Comment (dtolnay) [inline testing/tests/test_macros.rs]: Wait what? How does the old one compile?

```rust
enum DeEnum<B, C: /* Trait */, D> /* where D: Trait */ {
```

without comments:

```rust
enum DeEnum<B, C: , D> {
```

Comment (dtolnay) [inline testing/tests/test_macros.rs]: Filed https://github.com/rust-lang-nursery/rustfmt/issues/1095

Comment (dtolnay) [inline testing/tests/test_macros.rs]: Also filed https://github.com/rust-lang-nursery/rustfmt/issues/1096 related to this line.

I bet all these comments are remnants of an erickt debugging session so we need to clean them up anyway.

File: serde/src/bytes.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ pub use self::bytebuf::{ByteBuf, ByteBufVisitor};
 #[cfg(feature = "collections")]
 use collections::Vec;
 
-///////////////////////////////////////////////////////////////////////////////
+/// ////////////////////////////////////////////////////////////////////////////
```
Comment (jwilm) [inline serde/src/bytes.rs]: This seems undesirable; the long string of `/` will show up in the docs.

Comment (dtolnay) [inline serde/src/bytes.rs]: As I noted in the comment, I already filed a rustfmt issue about this.

Comment (jwilm) [inline serde/src/bytes.rs]: Ah, sorry. I didn't actually jump to the rustfmt issue links.

**Comment by erickt** [comment]
@dtolnay: Yeah I thought about those aster chains being a problem for rustfmt. One way of handling this we could change the aster api to use closures instead of doing my silly chaining. So your example before could look like:

```rust
    let wrapper_ty = builder.path()
        .segment("__SerializeWith", |b| {
            b.with_generics(wrapper_generics.clone()).build()
        })
        .build();
```

Which could look much nicer. Or we just lift these segments out into a variable like:

```rust
    let segment = builder.segment("__SerializeWith")
        .with_generics(wrapper_generics.clone())
        .build();
    let wrapper_ty = builder.path()
        .with_segment(segment)
        .build();
```

Or even just create the variable in place with:

```rust
    let wrapper_ty = builder.path()
        .with_segment(
            builder.segment("__SerializeWith")
                .with_generics(wrapper_generics.clone())
                .build()
        )
        .build();
```

**Pull request closed without merging**